x-pack/plugins/security_solution/public/common/components/alerts_treemap_panel/index.test.tsx

```diff
@@ -0,0 +1,247 @@
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() =>
+      expect(screen.getByTestId('treemapPanel')).toHaveStyleRule('overflow-x', 'hidden')
+    );
+  });
+
+  it('renders the panel with an auto overflow-y to allow vertical scrolling when necessary', async () => {
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() =>
+      expect(screen.getByTestId('treemapPanel')).toHaveStyleRule('overflow-y', 'auto')
+    );
+  });
+
+  it('renders the chart selector as a custom header title', async () => {
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() => expect(screen.getByTestId('chartSelect')).toBeInTheDocument());
+  });
+
+  it('renders field selection when `isPanelExpanded` is true', async () => {
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() => expect(screen.getByTestId('fieldSelection')).toBeInTheDocument());
+  });
+
+  it('does NOT render field selection when `isPanelExpanded` is false', async () => {
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} isPanelExpanded={false} />
+      </TestProviders>
+    );
+
+    await waitFor(() => expect(screen.queryByTestId('fieldSelection')).toBeNull());
+  });
+
+  it('renders the progress bar when data is loading', async () => {
+    (useQueryAlerts as jest.Mock).mockReturnValue({
+      loading: true,
+      data: mockAlertSearchResponse,
+      setQuery: () => {},
+      response: '',
+      request: '',
+      refetch: () => {},
+    });
+
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() => expect(screen.getByTestId('progress')).toBeInTheDocument());
+  });
+
+  it('does NOT render the progress bar when data has loaded', async () => {
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() => expect(screen.queryByTestId('progress')).toBeNull());
+  });
+
+  it('renders the treemap when data is available and `isPanelExpanded` is true', async () => {
+    jest.mock('../../../detections/containers/detection_engine/alerts/use_query', () => {
+      return {
+        useQueryAlerts: () => ({
+          loading: true,
+          data: mockAlertSearchResponse,
+          setQuery: () => {},
+          response: '',
+          request: '',
+          refetch: () => {},
+        }),
+      };
+    });
+
+    render(
+      <TestProviders>
+        <AlertsTreemapPanel {...defaultProps} />
+      </TestProviders>
+    );
+
+    await waitFor(() =>
+      expect(screen.getByTestId('treemap').querySelector('.echChart')).toBeInTheDocument()
```

Comment:
Is there a label or another test id that could be used instead of `querySelector('.echChart')`?